pandas/core/tools/numeric.py

```diff
@@ -188,6 +196,15 @@ def to_numeric(arg, errors="raise", downcast=None):
                     if values.dtype == dtype:
                         break
 
+    if mask is not None:
+        data = np.zeros(mask.shape, dtype=values.dtype)
+        data[~mask] = values
+
+        from pandas.core.arrays import FloatingArray, IntegerArray
```

Comment:
this has to be here and not at module-level import because pandas.core.arrays.integer uses `to_numeric`